api/src/main/java/org/openmrs/module/openconceptlab/OclConceptService.java

```diff
@@ -0,0 +1,23 @@
+package org.openmrs.module.openconceptlab;
+
+import org.openmrs.Concept;
+import org.openmrs.annotation.Authorized;
+import org.openmrs.util.PrivilegeConstants;
+import org.springframework.transaction.annotation.Transactional;
+
+public interface OclConceptService {
+	
+	/**
+	 * A version of {@link org.openmrs.api.ConceptService#getConceptByMapping(String, String)}
+	 * which returns concepts that are marked as the "SAME-AS" an existing concept
+	 *
+	 * @param source the name of the source to check
+	 * @param conceptId the identifier of the concept
+	 * @return the concept that is semantically the same as the requested concept or null if
+	 *  one couldn't be found.
+	 */
+	@Transactional(readOnly = true)
+	@Authorized(PrivilegeConstants.VIEW_CONCEPTS)
+	Concept getDuplicateConceptByMapping(String conceptId, String source);
```

Comment:
I find this method name confusing.  Can we just call it "getConceptWithSameAsMapping"?  And name "conceptId" -> "code"?

Reply:
Well, the name arose because I was planning on eventually including Burke's "CODE" mapping if / when that comes about, but yeah, I suppose that's more accurate.